api/lifecycle.go

```diff
@@ -0,0 +1,32 @@
+// NOTE: Subject to change, do not rely on this package from outside git-lfs source
+package api
+
+import "net/http"
+
+// TODO: extract interface for *http.Request; update methods. This will be in a
+// later iteration of the API client.
+
+// A Lifecycle represents and encapsulates the behavior on an API request from
+// inception to cleanup.
+//
+// At a high level, it turns an *api.RequestSchema into an
+// api.Response (and optionally an error). Lifecycle does so by providing
+// several more fine-grained methods that are used by the client to manage the
+// lifecycle of a request in a platform-agnostic fashion.
+type Lifecycle interface {
+	// Build creates a sendable request by using the given RequestSchema as
+	// a model.
+	Build(req *RequestSchema) (*http.Request, error)
+
+	// Execute transforms generated request into a wrapped repsonse, (and
+	// optionally an error, if the request failed), and serializes the
+	// response into the `into interface{}`, if one was provided.
+	Execute(req *http.Request, into interface{}) (Response, error)
```

Comment:
You're right, but I had mentioned to @technoweenie that I thought this was a good intermediate step between the old HTTP-specific code and the new API client stuff. Eventually, I'd like to find the correct abstraction between the `*http.Request` and `api.Request`, but until we know what the SSH API is going to look like, this is difficult.

If it's okay with you, I think we should leave this and the `Method` field on `RequestSchema` as is until we can see what an SSH api would need specifically.